discretize/_extensions/tree_ext.pyx

```diff
@@ -6964,6 +6998,19 @@ cdef class _TreeMesh:
         """
         return self.get_cells_in_aabb(*rectangle.reshape(self.dim, 2).T)
 
+    def _error_if_not_finalized(self, method: str, is_property: bool):
+        """
+        Raise error if mesh is not finalized.
+        """
+        object_name = "property" if is_property else "method"
+        if not self.finalized:
+            msg = (
+                f"The `{method}` {object_name} "
+                "cannot be accessed before the mesh is finalized. "
+                "Use the `finalize()` method to finalize it."
+            )
+            raise AttributeError(msg)
+
```

Comment:
I don't think saying that the item is a property or not is necessary, something along the lines of:

```suggestion
        if not self.finalized:
            msg = (
                f"`{type(self).__name__}.{method}` requires a finalized mesh. "
                "Use the `finalize()` method to finalize it."
            )
            raise TreeMeshNotFinalizedError(msg)

```